src/plot_api/plot_api.js

```diff
@@ -302,7 +303,7 @@ Plotly.plot = function(gd, data, layout, config) {
                     .selectAll(query)
                     .remove();
 
-                fullLayout._infolayer.selectAll('g.rangeslider-container')
+                rangesliderContainers
```

Comment:
Did not tested separately. Just checked and my tests shows 200ms to 400ms improvement on 200 traces.

When I profile the 200-trace case, the ``select`` and ``selectAll`` are among top 5 of hot functions. So my thought was caching all the constant selected elements in data loops.